lib/common/alarm/AlarmFactory.ts

```diff
@@ -821,9 +821,37 @@ export class AlarmFactory {
     };
   }
 
+  /**
+   * Creates a composite alarm from a collection of alarms and composite alarms.
+   *
+   * @param alarms Array of individual alarms to be composed
+   * @param props Customization options for the composite alarm
+   * @param compositeAlarms Optional array of composite alarms to be composed together with regular alarms
+   * @returns Newly created composite alarm
+   */
   addCompositeAlarm(
     alarms: AlarmWithAnnotation[],
     props: AddCompositeAlarmProps,
+    compositeAlarms: CompositeAlarm[] = [],
+  ): CompositeAlarm {
+    const alarmBases = [
+      ...alarms.map((alarm) => alarm.alarm),
+      ...compositeAlarms,
+    ] as AlarmBase[];
+    return this.addCompositeAlarmFromAlarmBases(alarmBases, props);
+  }
+
+  /**
+   * Creates a composite alarm from a collection of alarm base objects.
+   * This allows creation of composite alarms that include both metric alarms and other composite alarms.
+   *
+   * @param alarms Array of AlarmBase objects (can include both Alarm and CompositeAlarm)
+   * @param props Customization options for the composite alarm
+   * @returns Newly created composite alarm
+   */
+  addCompositeAlarmFromAlarmBases(
```

Comment:
I wonder if the new method should also return an `AlarmWithAnnotation` like `addAlarm` does for better consistency?

Also not a fan of this method name, but nothing better immediately comes to mind. :\

Reply:
Yeah, I'm not a fan of the names either. I wish I could just overload the method, but that's not an option here.

> I wonder if the new method should also return an `AlarmWithAnnotation`, like `addAlarm` does, for better consistency?

I'm not sure I follow. The existing method `addCompositeAlarm()` already returns a `CompositeAlarm`. Wouldn't we want the new method to align with that?